### ISSUE TITLE
Provide a button for admin user(s) to re-index all files of all users

### ISSUE THREAD

**Issue by jerabaul29** [nextcloud/server#35816]
I quite often encounter the situation where the index of all files is not up to date (usually, because there was a "direct upload" taking place one way or another to the data folder of some user). I understand that it is possible to use the ```occ``` command on the server command line side to re-run a full indexing (and there are many hits on google for this). However, doing things "the occ way" is a bit of a hazzle, since doing things right may depend on how things are installed: which php user to use, what path nextcloud is installed into, is it running natively or in docker, etc.

Hence my question: would it be possible that admins get a "refresh the full index" button that they can click in the nextcloud web interface, that re-indexes all files for all users (if possible, it would be great to be able to re-build for specific users / paths too with a selection menu, but guess this is more work, for a first iteration "just" rebuild all would be already of great help :) ), in the correct way, without the need to do any more work of finding right options etc?

**Comment by tobiasKaminsky** [comment]
This fits better in server.

Maybe this helps: https://apps.nextcloud.com/apps/occweb
This way you can run occ directly from web.
(never tried it…)

**Comment by jerabaul29** [comment]
Nice, thanks for the link to the occweb extension. So this will use the right occ parameters so that the command is correctly run, right? :)

**Comment by kesselb** [comment]
There is a background job to regularly run a background scan for each user.
To trigger a scan via `occ files:scan`.

> would it be possible that admins 

Possible yes. Either by enqueue the background scan or trigger the scan logic via browser. A scan via browser has to deal with timeouts by the webserver. 

> However, doing things "the occ way" is a bit of a hazzle, since doing things right may depend on how things are installed: which php user to use, what path nextcloud is installed into, is it running natively or in docker, etc.

Our documentation has some pointers for occ: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/occ_command.html

Currently there are no plans to implement such a feature. Thus I will close this ticket for now. This does not mean we don't want this feature, but it is simply not on our roadmap for the near future. If somebody wants to implement this feature nevertheless we are happy to assist and help out.

**Comment by jerabaul29** [comment]
To be honest, this feature would be really needed in my case. Somehow I do not manage to upload large amounts of pictures / documents through the android app, and the only reliable way I have found is to scp the data to the data folder. However, without re-indexing, the data are not visible in the android app. Looks like I am not the only user having issues with the android app, see for example the discussions at https://www.reddit.com/r/NextCloud/comments/php0n6/android_app_is_really_sluggish/ https://github.com/nextcloud/android/issues/3692 https://news.ycombinator.com/item?id=18648023 . I tried to understand why I have issues with it, but never managed to find the cause. If there would be a re-index button available, that would mean that I could use the ssh/scp-ing method, and re-index, and then all would be fine to me.
